R/verb-summarise.R

```diff
@@ -66,8 +113,21 @@ op_sort.op_summarise <- function(op) NULL
 #' @export
 sql_build.op_summarise <- function(op, con, ...) {
   select_vars <- translate_sql_(op$dots, con, window = FALSE, context = list(clause = "SELECT"))
-  group_vars <- c.sql(ident(op_grps(op$x)), con = con)
+  group_vars <- op_grps(op$x)
+  n <- length(group_vars)
+
+  .groups <- op$args$.groups
+  verbose <- summarise_verbose(.groups, caller_env())
+  # we cannot automatically infer "keep" as for local frames
```

Comment:
Why would we infer `keep`? I think I've forgotten something.

Reply:
From the documentation of `summarise`:

> When .groups is not specified, it is chosen based on the number of rows of the results:
> * If all the results have 1 row, you get "drop_last".
> * If the number of rows varies, you get "keep".

For local dataframes the check is easy but for db tables we cannot easily check that (I think it is not even possible that multiple rows are returned per group but).

Reply:
Yeah, in databases `summarise()` always returns a single row.